src/helper-modules/companies-house/companies-house.service.ts

```diff
@@ -0,0 +1,48 @@
+import { HttpService } from '@nestjs/axios';
+import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { CompaniesHouseConfig, KEY as COMPANIES_HOUSE_CONFIG_KEY } from '@ukef/config/companies-house.config';
+import { HttpClient } from '@ukef/modules/http/http.client';
+
+import { GetCompanyCompaniesHouseResponse } from './dto/get-company-companies-house-response.dto';
+import {
+  getCompanyInvalidAuthorizationKnownCompaniesHouseError,
+  getCompanyMalformedAuthorizationHeaderKnownCompaniesHouseError,
+  getCompanyNotFoundKnownCompaniesHouseError,
+} from './known-errors';
+import { createWrapCompaniesHouseHttpGetErrorCallback } from './wrap-companies-house-http-error-callback';
+
+@Injectable()
+export class CompaniesHouseService {
+  private readonly httpClient: HttpClient;
+  private readonly key: string;
+
+  constructor(httpService: HttpService, configService: ConfigService) {
+    this.httpClient = new HttpClient(httpService);
+    const { key } = configService.get<CompaniesHouseConfig>(COMPANIES_HOUSE_CONFIG_KEY);
+    this.key = key;
+  }
+
+  async getCompanyByRegistrationNumber(registrationNumber: string): Promise<GetCompanyCompaniesHouseResponse> {
```

Comment:
As far as I'm aware, Typescript does not have this feature (if we throw rather than return an exception): https://www.learningtypescript.com/articles/why-typescript-doesnt-include-a-throws-keyword

Reply:
You should be able to do `Promise<GetCompanyCompaniesHouseResponse| getCompanyMalformedAuthorizationHeaderKnownCompaniesHouseError | getCompanyNotFoundKnownCompaniesHouseError...>`

Reply:
From my understanding, `Promise<GetCompanyCompaniesHouseResponse | getCompanyMalformedAuthorizationHeaderKnownCompaniesHouseError | getCompanyNotFoundKnownCompaniesHouseError...>` would be appropriate if the function *returned* those errors, but instead it throws them. See the `Union Types` section in https://www.learningtypescript.com/articles/why-typescript-doesnt-include-a-throws-keyword